lib/ansible/modules/system/firewalld.py

```diff
@@ -926,6 +977,27 @@ def main():
         changed, transaction_msgs = transaction.run()
         msgs = msgs + transaction_msgs
 
+    ''' If there are no changes within the zone we are operating on the zone itself '''
+    if modification_count == 0 and desired_state in ['absent', 'present']:
+
+        transaction = ZoneTransaction(
+            fw,
+            action_args=(),
+            zone=zone,
+            desired_state=desired_state,
+            permanent=True,
+            immediate=False,
+            fw_offline=fw_offline
+        )
```

Comment:
I'd like to see the `NotImplementedError` exceptions be handled here so the user gets a clean error message via `module.fail_json` in the event one of those occurs.